src/sentry/models/dashboard_permissions.py

```diff
@@ -3,10 +3,23 @@
 from django.db import models
 
 from sentry.backup.scopes import RelocationScope
-from sentry.db.models import Model, region_silo_model
+from sentry.db.models import FlexibleForeignKey, Model, region_silo_model
 from sentry.db.models.base import sane_repr
 
 
+@region_silo_model
+class DashboardPermissionsTeam(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    team = FlexibleForeignKey("sentry.Team")
+    permissions = FlexibleForeignKey("sentry.DashboardPermissions")
```

Comment:
I believe for both of these fields we want to use an `on_delete=models.CASCADE`. When a team is deleted, we want the corresponding entries to be deleted (i.e. only the teams that exist are still assigned as permissions in the dashboard). And when the permission is deleted, we should delete all of the leftover relationships between that permission and the teams